### ISSUE TITLE
Neither HHVM nor typechecker support trailing commas after variadics

### ISSUE THREAD

**Issue by fredemmott** [facebook/hhvm#7657]
### HHVM Version

3.17

### Standalone code, or other way to reproduce the problem

```Hack
<?hh

class Foo {
  public function bar (
    string $herp,
    mixed ...$derp,
  ): void {
  }

  public function baz(): void {}
}
```

### Expected result

No problems.

### Actual result

Runtime: `Fatal error: syntax error, unexpected ',', expecting ')' in /private/tmp/hhtest/test.php on line 6` - fairly clear

Typechecker: `test.php:4:19,21: This method is not declared as abstract, it must have a body (NastCheck[3014])
test.php:6:19,19: Expected ) (Parsing[1002])` (very misleading)

The error the typechecker raises varies wildly depending on the context - in some cases, this *only* shows up as  "Unbound Name: someclass" errors - I'm preliminary putting on mid-pri because of this.

**Comment by fredemmott** [comment]
`hackfmt` generates code that is invalid in this way.

**Comment by fredemmott** [comment]
Removed probably easy tag: the actual code change is likely very easy, the hard part is we don't have consensus on if this is a bug or not:

Argument for bug: it's inconsistent and unexpected

Argument for desired behavior: you can't have additional arguments after a variadic, so a trailing comma doesn't make sense

**Comment by fredemmott** [comment]
hackfmt fixed by 035aed7c9e43ae854d10e42a8fb2b0f484938997, but still no consensus on correct behavior.

**Comment by lexidor** [comment]
I am going over old issues on this repository, to see which ones apply to the current versions of hhvm.

The error message has been updated to address the issue directly.
I would assume that this means this is intentional.

```
Parsing[1002] A variadic parameter ('...') may not be followed by a comma.
   --> src/file.hack
  7 |   public function bar(string $herp, mixed ...$derp,): void {
    |                                                   ^

1 error found.
```
